### ISSUE TITLE
Coordinating Completion of Organic Groups

### ISSUE THREAD

**Issue by stpaultim** [backdrop-contrib/og#30]
In a very interesting and engaging discussion about Outreach efforts for Backdrop CMS, a couple of community members raised the issue of getting Organic Groups ported to Backdrop and questioned what could be done to expedite this issue. They suggested we talk about this issue during our weekly dev meeting, which we did this afternoon. 

Some of the issues that came up or points that were made:

1) Is this issue a blocker: https://github.com/backdrop/backdrop-issues/issues/3646 - It was suggested that this issue is a blocker for Organic Groups. @quicksketch suggested that probably shouldn't be a blocker, that there are ways of working around this issue. @jenlampton commented: https://github.com/backdrop/backdrop-issues/issues/3646#issuecomment-1023656522

2) @olafgrabienski mentioned that he has clients who are dependent upon this module that are currently making decisions about upgrading from Drupal 7 to Backdrop or some other platform and that this module might be a deciding factor. 

3) We discussed the following strategies that anyone can take to help shepherd this module towards completion. 

 - Organize a session at Backdrop LIVE to plan and stratagize
 - [Become an advocate](https://backdropcms.org/news/become-an-advocate-for-a-new-feature-in-backdrop-core) for any Core issues that might be blocking it.
 - Be active in this issue queue - asking questions or gently prodding the community to action
 - Try to identify other organizations interested in this module and see if a contribution sprint might be possible to focus on moving this project forward
 - Organize a "GoFundMe" or "Kickstarter" to raise enough money for a bounty to move things along

4) Check in with @laryn (currently maintainer of this repo) to see what his plans are (without putting pressure on @laryn to do all the work). 

@laryn - is this something that still interests you and/or do you imagine having time in the near future for this project?

NEXT STEPS:

1) Personally, I'm very interested in this module and might be willing to contribute some developer time or a small financial contribution towards this project. If anyone else is in a similar situation, let us know. 

2) We can talk here or in Zulip.

3) Decide if a meeting would be helpful to reinvigorate this project and/or plan a strategy?

**Other places this is being discussed:**

Forum: https://forum.backdropcms.org/forum/organic-groups-module-backdrop
Port Requests: https://github.com/backdrop-ops/contrib/issues/420

**Comment by olafgrabienski** [comment]
Thanks for opening the issue, Tim! One thing to add: the Translation server (localize.backdropcms.org) is running on D7 and Organic Groups. So, making Organic Groups ready for Backdrop would also help to upgrade the Translation server to Backdrop.

**Comment by docwilmot** [comment]
When I get some time I'm going to see if I can get tests to pass. I find if youve got comprehensive tests (this seems to), if you can get them to green, you're almost perfect.

**Comment by laryn** [comment]
Getting this module going is definitely still of interest but I'd love one or more co-maintainers if they are available. I don't currently have a project that requires OG so my time is a bit limited. I saw [the note after the dev meeting regarding the Configurable Actions](https://github.com/backdrop-contrib/og/issues/22#issuecomment-1024687424) which could help get this unstuck though!

**Comment by stpaultim** [comment]
Does anyone think that organizing either a strategy meeting or sprint day would be helpful. 

1) The goal of a strategy meeting would be to try to get a sense of the following.

  a - How far have we gotten so far (how far has @laryn gotten)?
  b - Do we have any idea how much work is left (this may be impossible to predict)?
  c - Do we have folks with the dev skills ready to work on it? Right now, it seems like @docwilmot is willing to give it a go. Is this enough or will he need help? Is there something we can do to recruit others to help?
  d - Would a little (limited) fundraising help so that those unable to write the code have a way of contributing. Maybe a small bounty ($500-$1000) would help incentivise someone with skills to commit time to the project.
  
 2) A sprint might help some us set aside time for this specific project and give those of us with limited coding skills the confidence to tackle some tasks with knowledge that someone is around to help us out or answer questions. 

Maybe we don't need either, just brainstorming. Also, thinking that anything we learn from this effort might be used to tackle other important core or contrib issues.

**Comment by olafgrabienski** [comment]
> Does anyone think that organizing either a strategy meeting or sprint day would be helpful?

In my opinion, a meeting would be helpful: to get a better impression of the module status, and to discuss different ways to contribute.

**Comment by VasasA** [comment]
> Do we have folks with the dev skills ready to work on it?

I am ready to work on it. :)

**Comment by laryn** [comment]
I reinstalled the current status of OG to see where we left it and there were some minor fixes needed to get it back to where I vaguely remembered it being (which were due to classes being renamed in EntityReference). So it should be installable and somewhat usable. At least further testable.

@VasasA Does this workaround/clue give you a place you feel comfortable starting/trying to get the configurable actions running? https://github.com/backdrop-contrib/og/issues/22#issuecomment-1024687424

**Comment by VasasA** [comment]
Sorry, but I don't know how can I get the configurable actions running. I installed the module, and I'm watching [this tutorial](http://modulesunraveled.com/organic-groups-7x-2x), and trying the steps.

**Comment by laryn** [comment]
@VasasA No problem, I may have misunderstood your earlier comment. Thanks for installing and testing -- feedback on your experience will be helpful!

**Comment by olafgrabienski** [comment]
> So it should be installable and somewhat usable. At least further testable.

Having a use case from a D7 site, I'll test and try to give helpful feedback.

**Comment by stpaultim** [comment]
@VasasA Posted this update in Zulip. I think it would be good document the status here and post periodic updates here, especially if there is a need for testing or some new development.

"The conversion of OG module is in progress. I would like to summarize the current situation:

1. Porting OG is very complex. For example: I fixed the [issue 11](https://github.com/backdrop-contrib/og/issues/11), but the solution was in the Entity Plus module, not in the OG: [issue 119](https://github.com/backdrop-contrib/entity_plus/issues/119) So you have to use the dev version of Entity Plus module if you install the OG.
2. Porting of Organic Groups module is hung up on a core issue. The [last comment of Jen](https://github.com/backdrop/backdrop-issues/issues/3646#issuecomment-1023656522) in the issue is important: Laryn is working on this work-around in a [new branch](https://github.com/backdrop-contrib/og/tree/1.x-2.x-issue-22).
3. (The issue of this problem in the OG issue queue: [issue 22](https://github.com/backdrop-contrib/og/issues/22) )
4. I am working on [the issue 35](https://github.com/backdrop-contrib/og/issues/35) now.
5. It is a useful [tutorial.](http://modulesunraveled.com/organic-groups)
6. We talked about OG and other important modules on [this dev meeting](https://www.youtube.com/watch?v=g3iGZWyaBOk) from 5:00 to 27:10"

**Comment by olafgrabienski** [comment]
I've reserved a time slot for Organic Groups at Backdrop Live (Block E, Slot 14, 5 pm UTC - March 12, 2022): https://events.backdropcms.org/sessions/organic-groups-past-present-future

I'm willing to facilitate the topic, if my internet connection allows it (will be on a vacation in those days). I appreciate any input and ideas from all of you.

**Comment by stpaultim** [comment]
Can someone post a quick update here on progress.

Is there any need for more testing yet?

**Comment by laryn** [comment]
I set up a project board for this issue in the hopes it might help keep things organized. Is anyone willing to try to organize existing issues related to this port there?

https://github.com/backdrop-contrib/og/projects/1

I'll add a few issues in there. I think I can set up some basic automation as well so closed issues move to the "Done" column, etc.

**Comment by olafgrabienski** [comment]
> Is there any need for more testing yet?

I've started rebuilding the Organic Groups setup from a client's D7 site in Backdrop. That was interesting, but I think I'll restart and use the tutorial from https://modulesunraveled.com/organic-groups-7x-2x as example. Using the tutorial, I expect it'll be more easy to discuss testing results.

**Comment by olafgrabienski** [comment]
I've started with the Drupal 7 Organic Groups tutorial, just tested the following chapters in Backdrop:

2. Organic Groups Setup
3. Organic Groups Views
4. Organic Groups Layout
5. Configuring Content Types for group "Content create links"

So far, most things work in Backdrop, but there are some differences to Drupal, and due to a missing contrib module I had to find a workaround for the "Create content links". Now I'm considering where to put my notes: in the Wiki or create a documentation issue thread? I prefer the latter as the notes are still work in progress.

**Comment by laryn** [comment]
Follow your preference here @olafgrabienski -- especially since the port is still very much in progress as well as the notes!

**Comment by olafgrabienski** [comment]
Yep, here we are: [Testing Organic Groups and adding notes](https://github.com/backdrop-contrib/og/issues/45)

**Comment by stpaultim** [comment]
I've been told that this module should be fairly stable at this point and that it primarily only needs testing. 

Maybe we'll be ready for a beta release soon, which should help generate additional testing.

**Comment by argiepiano** [comment]
Thank you all for this conversation and your work getting the port moving forward. Most of the issues here have been solved in the last month. I'm closing issues that are not current.

A quick note about configurable actions needed in one of the OG views: that was solved by using Backdrop core's Bulk Operations, and these actions are up and running. Later, after this fix, I ported VBO, but that module is not needed in the current version of Backdrop OG - core actions can be made into configurable ones, which is what I did in #22.

We are getting ready for a beta release in the next day or two.
